### ISSUE TITLE
get_settings() mutates DEFAULT_SETTINGS

### ISSUE THREAD

**Issue by ted537** [Vennify-Inc/chatting-transformer#13]
VIew `num_beams` in this example. After the settings for beam search have been fetched, the settings for greedy are incorrect.

```
>>> from chattingtransformer.settings import *
>>> print(get_settings("greedy",None,None))
{'max_length': 60, 'min_length': 10, 'do_sample': False, 'early_stopping': False, 'num_beams': 1, 'temperature': 0.65, 'top_k': 50, 'top_p': 1.0, 'repetition_penalty': 1, 'length_penalty': 1, 'no_repeat_ngram_size': 2, 'bad_words_ids': None}
>>> print(get_settings("beam-search",None,None))
{'max_length': 60, 'min_length': 10, 'do_sample': False, 'early_stopping': True, 'num_beams': 5, 'temperature': 0.65, 'top_k': 50, 'top_p': 1.0, 'repetition_penalty': 1, 'length_penalty': 1, 'no_repeat_ngram_size': 2, 'bad_words_ids': None}
>>> print(get_settings("greedy",None,None))
{'max_length': 60, 'min_length': 10, 'do_sample': False, 'early_stopping': True, 'num_beams': 5, 'temperature': 0.65, 'top_k': 50, 'top_p': 1.0, 'repetition_penalty': 1, 'length_penalty': 1, 'no_repeat_ngram_size': 2, 'bad_words_ids': None}
>>> 
```

**Comment by EricFillion** [comment]
Thanks! I fixed it in #16 by adding  ".copy()" when creating the settings variable from the DEFAULT_SETTINGS.

**Comment by EricFillion** [comment]
Resolved in version 1.0.1.
